saleor/graphql/channel/mutations/channel_update.py

```diff
@@ -82,9 +88,27 @@ def clean_input(cls, info, instance, data, input_cls=None):
             cleaned_input["slug"] = slugify(slug)
         if stock_settings := cleaned_input.get("stock_settings"):
             cleaned_input["allocation_strategy"] = stock_settings["allocation_strategy"]
+        if order_settings := cleaned_input.get("order_settings"):
+            cleaned_input["automatically_confirm_all_new_orders"] = order_settings[
+                "automatically_confirm_all_new_orders"
+            ]
+            cleaned_input[
+                "automatically_fulfill_non_shippable_gift_card"
+            ] = order_settings["automatically_fulfill_non_shippable_gift_card"]
 
         return cleaned_input
 
+    @classmethod
+    def check_permissions(cls, context, permissions=None, **data):
+        permissions = [ChannelPermissions.MANAGE_CHANNELS]
+
+        # when updating only orderSettings allow MANAGE_ORDERS permission
+        input = data["data"]["input"]
+        if "order_settings" in input and len(input) == 1:
```

Comment:
Why `and len(input) == 1`? 
If I have dict like below that mean Saleor shouldn't require `MANAGE_ORDERS`? 
```
input = {
    "stock_settings": ...,
    "order_settings":...,
}
````

Reply:
If I'm right you should add a unit test for this situation.

Reply:
Yes, `MANAGE_ORDERS` permission can be used `only` when we are updating `order_settings`.
And I already added tests for this case.